ui/base/src/main/java/org/cru/godtools/base/ui/databinding/YouTubePlayerViewBindingAdapter.kt

```diff
@@ -11,6 +11,14 @@ import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.utils.YouTube
 import com.pierfrancescosoffritti.androidyoutubeplayer.core.player.views.YouTubePlayerView
 import org.cru.godtools.base.ui.R
 
+@BindingAdapter("pauseVideoWhen")
+internal fun YouTubePlayerView.pauseVideoWhen(pause: Boolean) {
```

Comment:
So before you added this, would it continue to play the video even if you moved on to the next page?

Reply:
yes and no.

last week the video would have stopped because the RecyclerView only kept the current page rendered, this caused other UX issues because images wouldn't be initially loaded when you navigated to a page for the first time.

To address that issue I increased the number of adjacent pages to pre-render. This caused the video to continue to play while it is on an adjacent page because it wasn't being destroyed immediately

I had used the AndroidX Lifecycle framework elsewhere to work around this issue, so I did the work to instrument the lesson renderer to use the lifecycle framework and keep it consistent with the other renderers